sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -3,479 +3,624 @@
-    messageId?: string | number | Buffer;
-    partitionKey?: string;
-    replyTo?: string;
-    replyToSessionId?: string;
-    scheduledEnqueueTimeUtc?: Date;
-    readonly sequenceNumber?: Long;
-    sessionId?: string;
-    timeToLive?: number;
-    to?: string;
-    userProperties?: {
-        [key: string]: any;
-    };
-    viaPartitionKey?: string;
+  dataTransformer?: DataTransformer;
+  webSocketOptions?: WebSocketOptions;
+}
+
+// @public
+export class ServiceBusMessage implements ReceivedMessageInfo {
+  abandon(propertiesToModify?: { [key: string]: any }): Promise<void>;
+  readonly _amqpMessage: AmqpMessage;
+  body: any;
+  clone(): SendableMessageInfo;
+  complete(): Promise<void>;
+  contentType?: string;
+  correlationId?: string | number | Buffer;
+  deadLetter(options?: DeadLetterOptions): Promise<void>;
+  readonly deadLetterSource?: string;
+  defer(propertiesToModify?: { [key: string]: any }): Promise<void>;
+  readonly delivery: Delivery;
+  readonly deliveryCount?: number;
+  readonly enqueuedSequenceNumber?: number;
+  readonly enqueuedTimeUtc?: Date;
+  readonly expiresAtUtc?: Date;
+  get isSettled(): boolean;
+  label?: string;
+  lockedUntilUtc?: Date;
+  readonly lockToken?: string;
+  messageId?: string | number | Buffer;
+  partitionKey?: string;
+  replyTo?: string;
+  replyToSessionId?: string;
+  scheduledEnqueueTimeUtc?: Date;
+  readonly sequenceNumber?: Long;
+  sessionId?: string;
+  timeToLive?: number;
+  to?: string;
+  userProperties?: {
+    [key: string]: any;
+  };
+  viaPartitionKey?: string;
 }
 
 // @public
 export interface SessionMessageHandlerOptions {
-    autoComplete?: boolean;
-    maxConcurrentCalls?: number;
-}
-
-// @public
-export class SessionReceiver {
-    close(): Promise<void>;
-    getMessageIterator(): AsyncIterableIterator<ServiceBusMessage>;
-    getState(): Promise<any>;
-    get isClosed(): boolean;
-    isReceivingMessages(): boolean;
-    peek(maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
-    peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
-    receiveDeferredMessage(sequenceNumber: Long): Promise<ServiceBusMessage | undefined>;
-    receiveDeferredMessages(sequenceNumbers: Long[]): Promise<ServiceBusMessage[]>;
-    receiveMessages(maxMessageCount: number, maxWaitTimeInSeconds?: number): Promise<ServiceBusMessage[]>;
-    get receiveMode(): ReceiveMode;
-    registerMessageHandler(onMessage: OnMessage, onError: OnError, options?: SessionMessageHandlerOptions): void;
-    renewSessionLock(): Promise<Date>;
-    get sessionId(): string | undefined;
-    get sessionLockedUntilUtc(): Date | undefined;
-    setState(state: any): Promise<void>;
-    }
+  autoComplete?: boolean;
+  maxConcurrentCalls?: number;
+}
+
+// @public
+export interface SessionReceiver<LockModeT extends "peekLock" | "receiveAndDelete"> {
+  // (undocumented)
+  close(): Promise<void>;
+  close(): Promise<void>;
+  diagnostics: {
```

Comment:
`diagnostics` might not be the right name but the structure could be okay.

Alternatives:
- mangle the name (and leave at root)

Reply:
Leave at the root and rename `peek` to `viewMessages` or `browseMessages` ?

Reply:
Sticking with this for now. No other names have really come forward for preview.1